src/Agent.Worker/TestResults/JunitResultReader.cs

```diff
@@ -311,8 +311,10 @@ private static TimeSpan GetTimeSpan(XmlNode rootNode, out bool TimeDataAvailable
                 var timeValue = rootNode.Attributes["time"].Value;
                 if (timeValue != null)
                 {
-                    double timeInSeconds = 0.0;
-                    if (Double.TryParse(timeValue, NumberStyles.Any, NumberFormatInfo.InvariantInfo, out timeInSeconds))
+                    // Ensure that the time data is a positive value within range
+                    if (Double.TryParse(timeValue, NumberStyles.Any, NumberFormatInfo.InvariantInfo, out double timeInSeconds) 
+                        && !Double.IsNaN(timeInSeconds) 
```

Comment:
you may wants to check for other two allowed value of infinity as well  NumberFormatInfo.PositiveInfinitySymbol, NumberFormatInfo.NegativeInfinitySymbol, or NumberFormatInfo.NaNSymbol